### ISSUE TITLE
HTTPS redirection affects ACME HTTP Challenge as well

### ISSUE THREAD

**Issue by klumbe** [traefik/traefik#7825]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

First of all: I am using Traefik with Docker (Compose) on a Raspi 4.

1. Created a CertrificateResolver to enable HTTPS with Letsencrypt inside the static configuration (traefik.yaml) using the HTTP (:80) EntryPoint
2. Created an HTTPS redirect (traefik.yaml)
3. Connected docker-containers with Host-rules
4. Restarted the containers

### What did you expect to see?
The CertResolver uses the HTTP-Challange on port 80 like it did with Traefik version 2.37.

### What did you see instead?
Traefik tried to get the certficate and failed because the http-request seems to be forwarded to HTTPS, Port 443 which is not publicly available (by intention).
See the log in the according section.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.4.0
Codename:     livarot
Go version:   go1.15.6
Built:        2021-01-19T17:26:51Z
OS/Arch:      linux/arm
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Static config (traefik.yaml):
```yaml
entryPoints:
        web:
                address: ":80"
                http:
                        redirections:
                                entryPoint: 
                                        to: websecure
                                        scheme: https
                                        permanent: true

        websecure:
                address: ":443"
                http:
                        tls:
                                certResolver: myresolver

providers:
        docker: {}

        file:
                filename: "/routers/routers.yml"
                watch: true

api:
        dashboard: true
        insecure: true

certificatesResolvers:
        myresolver:
                acme:
                        email: "email@example.org"
                        storage: "/letsencrypt/acme.json"
                        httpChallenge:
                                entryPoint: web
```

routers.yml which just assigns a host to the dashboard:
```yaml
http:
        routers:
                dashboard:
                        rule: Host(`traefik.local.example.org`)
                        service: api@internal
```

docker-compose.yml for Traefik:
```yaml
version: '3'

services:
        reverse-proxy:
                image: traefik:v2.4
                ports:
                        - "80:80"
                        - "443:443"
                        - "8080:8080"
                volumes:
                        - /var/run/docker.sock:/var/run/docker.sock:ro
                        - ./traefik.yaml:/etc/traefik/traefik.yaml
                        - ./letsencrypt:/letsencrypt
                        - ./routers.yml:/routers/routers.yml
                labels:
                        - "traefik.enable=true"

networks:
        default:
                external:
                        name: traefik-proxy
```

docker-compose.yml of an example container WHOAMI:
```yaml
version: '3'
services:
        whoami:
                image: traefik/whoami
                labels:
                        - "traefik.http.routers.whoami.rule=Host(`whoami.local.example.org`)"

networks:
        default:
                external:
                        name: traefik-proxy
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
reverse-proxy_1  | time="2021-01-29T15:09:25Z" level=error msg="Unable to obtain ACME certificate for domains \"whoami.local.example.org\": unable to generate a certificate for the domains [whoami.local.example.org]: error: one or more domains had a problem:\n[whoami.local.example.org] acme: error: 400 :: urn:ietf:params:acme:error:connection :: Fetching https://whoami.local.example.org/.well-known/acme-challenge/qGUkV7dTJjElsaGodjK1bgg8u1i4gQev29PHsQoYGmw: Timeout during connect (likely firewall problem), url: \n" providerName=myresolver.acme routerName=websecure-whoami@docker rule="Host(`whoami.local.example.org`)"

```

**Comment by jspdown** [comment]
Hello @klumbe and thanks for your interest in Traefik.

Since you are using `redirections` directly on the entrypoints, the behavior is perfectly normal. You might actually want to use the `redirect` middleware on a router instead. I'm going to close this issue as a `question`. If you need some help do not hesitate to ask questions on the [community forum](https://community.traefik.io/c/traefik/5).

**Comment by jspdown** [comment]
My bad, there might actually be a bug. I'm reopening the issue.

**Comment by klumbe** [comment]
> My bad, there might actually be a bug. I'm reopening the issue.

@jspdown : Thanks for checking. Just let me know if I should try something or you need any further information.
Right now I don't see how to realize a general redirect to https with the redirect middleware on the router-side, but this may be a lack of knowledge on my side. Wouldn't I need to do this for every single router / service then?

**Comment by traefiker** [comment]
Closed by #7851.

**Comment by klumbe** [comment]
Thanks for the very fast fix. Works like a charm with version 2.4.2
